presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/utils/StatementUtils.java

```diff
@@ -129,9 +129,9 @@ private StatementUtils() {}
         builder.put(DropView.class, QueryType.DATA_DEFINITION);
         builder.put(CreateMaterializedView.class, QueryType.DATA_DEFINITION);
         builder.put(DropMaterializedView.class, QueryType.DATA_DEFINITION);
-        builder.put(CreateFunction.class, QueryType.DATA_DEFINITION);
+        builder.put(CreateFunction.class, QueryType.CONTROL);
         builder.put(AlterFunction.class, QueryType.DATA_DEFINITION);
```

Comment:
I actually wonder if we should move AlterFunction as well to Control types?

Reply:
i thought about it, but it doesn't use session because there's no support for temporary functions.  It's only the temporary functions really that need the control type, but they're bundled up together.  Ideally temporary functions would be split out, but that's a much bigger change and wasn't worth the bother right now.